### PR TITLE
Fix `FindProvider` calls on `SkinProvidingContainer` not considering disable flags

### LINKED ISSUE #13394: Beatmap skin doesn't get completely deactivated when having it set to off
**Describe the bug:** 
When playing a beatmap that has it's own skin and having beatmap skins disabled, the cursor, moving object on sliders as well as the "connection lines" between hit objects still appear in the beatmaps skin.
**Screenshots or videos showing encountered issue:** 
https://drive.google.com/file/d/1DLuTEWBd2O9X_KZ6PMK_LqN3uaB7jQo8/view?usp=sharing
First map doesn't has it's own skin (and that's how the skin usually looks), second map has a skin which results in these visual glichtes.
**osu!lazer version:** 
2021.608.0
**Logs:**

### PR DESCRIPTION
**Pull request by peppy**
Can't find a better way to handle this. Did look at making overloaded `FindProvider` calls but some special cases we use require the predicate function (see https://github.com/ppy/osu/blob/2c1f22d7ae914e84a6263d4ae7231d2f0f2a3bb7/osu.Game/Skinning/LegacySkinExtensions.cs#L30-L37).

Closes #13394.

### CODE REVIEW

**Comment by bdach** [comment]
On checking the beatmap linked in the issue, things aren't 100% correct yet (besides the slider ball colour issue). The HP bar and the score counter don't update correctly, but I haven't yet looked in detail as to why.

**Comment by peppy** [comment]
Nice catch. There was a missing passing of `SourceChanged` through the transformer in the case where the transformer's main source is an `ISkinSource`. Have added this in, but also keep in mind that this flow is likely going to go away with https://github.com/ppy/osu/issues/13403.

**Review by bdach — APPROVED** [review]
seems good now :+1:

File: osu.Game/Skinning/SkinProvidingContainer.cs [hunk 1]
```diff
@@ -53,54 +60,79 @@ public ISkin FindProvider(Func<ISkin, bool> lookupFunction)
             }
             else if (skin != null)
             {
-                if (lookupFunction(skin))
+                // a proxy must be used here to correctly pass through the "Allow" checks without implicitly falling back to the fallbackSource.
+                if (lookupFunction(noFallbackLookupProxy))
```
Comment (smoogipoo) [inline osu.Game/Skinning/SkinProvidingContainer.cs:64]: I'm so confused. How does this in any way "pass through the Allow checks"?

```
noFallbackLookupProxy = new NoFallbackProxy(this);

noFallbackLookupProxy.GetTexture(...)
  == provider .GetTexture(..., false)
  == this.GetTexture(..., false)
```
What am I missing?

Comment (peppy) [reply osu.Game/Skinning/SkinProvidingContainer.cs:64]: by "pass through" i mean "run" them.

the critical part of that comment is the following words "without implicitly falling back to the fallbackSource"

**Review by smoogipoo — APPROVED** [review]

File: osu.Game/Skinning/SkinProvidingContainer.cs (hunk 1 above)
Comment (smoogipoo) [reply osu.Game/Skinning/SkinProvidingContainer.cs:64]: I'm not sure why, but having public methods with the fallback parameter confused me.

Comment (peppy) [reply osu.Game/Skinning/SkinProvidingContainer.cs:64]: Do you think it's worth tidying them up? Moving the interface implementation to implicit and giving `fallback` a default parameter?

Comment (smoogipoo) [reply osu.Game/Skinning/SkinProvidingContainer.cs:64]: Dunno, it's fine for now. Probably makes more sense this way from the consumer's POV.

**Pull request merged**